### PR TITLE
User can not change password when New Password and Confirm Password a…

### PR DESCRIPTION
**Pull request by HoanVanHuynh**
User can not change password when New Password and Confirm Password are different

### CODE REVIEW

File: src/main/java/page_objects/BasePage.java [hunk 1]
```diff
@@ -13,6 +13,7 @@
     private final By tabLogout = By.cssSelector("a[href$='Logout']");
     private final By tabContact = By.cssSelector("a[href*='Contact']");
     private final By tabRegister = By.cssSelector("li > a[href$='Register.cshtml']");
+    private final By tabChangePassword = By.cssSelector(" a[href*='ChangePassword']");
```
Comment (tanmle) [inline src/main/java/page_objects/BasePage.java]: why isn't this element consistent with the other tabs?

File: src/test/java/tests/RegisterTests.java [hunk 2]
```diff
@@ -40,4 +44,42 @@ public void tc07_RegisterWithValidInformation() {
...
+        changePasswordPage.clickChangePasswordTab();
+
+        String newPassword = "a123:\"/{}!@$\\";
+        String confirmPassword = "b456:\"/{}!@$\\";
+
+        LogHelper.info("Change password with new password and confirm password are different");
+        changePasswordPage.changePassword(password, newPassword, confirmPassword);
+
+        LogHelper.info("Get error message at the top of change password form ");
+        String actualError = changePasswordPage.getChangePasswordErrorMessage();
```
Comment (tanmle) [inline src/test/java/tests/RegisterTests.java]: why aren't these var names consistent with the confirm password ones?

**Review by tanmle — CHANGES_REQUESTED** [review]

File: src/main/java/page_objects/BasePage.java (hunk 1 above)
Comment (HoanVanHuynh) [reply src/main/java/page_objects/BasePage.java]: thank you, I'm so sorry, I will delete space before locator a[href*='ChangePassword']

File: src/test/java/tests/RegisterTests.java (hunk 2 above)
Comment (HoanVanHuynh) [reply src/test/java/tests/RegisterTests.java]: Thank you, I'll rename all those var names

File: src/test/java/tests/RegisterTests.java [hunk 3]
```diff
@@ -40,4 +44,42 @@ public void tc07_RegisterWithValidInformation() {
...
+        changePasswordPage.clickChangePasswordTab();
+
+        String newPassword = "a123:\"/{}!@$\\";
+        String confirmPassword = "b456:\"/{}!@$\\";
+
+        LogHelper.info("Change password with new password and confirm password are different");
+        changePasswordPage.changePassword(password, newPassword, confirmPassword);
+
+        LogHelper.info("Get error message at the top of change password form ");
+        String actualChangePasswordErrorAtTop = changePasswordPage.getPasswordErrorMessageAtTop();
```
Comment (MinhQuan119) [inline src/test/java/tests/RegisterTests.java]: Need to check with Hoan about what error is, variable name is so long. Please don't merge it yet

Comment (HoanVanHuynh) [reply src/test/java/tests/RegisterTests.java]: Thank you so much, I'll rename it

Comment (MinhQuan119) [reply src/test/java/tests/RegisterTests.java]: image

In this case, it can be actualFormErrorMsg

File: src/test/java/tests/RegisterTests.java [hunk 4]
```diff
@@ -40,4 +44,42 @@ public void tc07_RegisterWithValidInformation() {
...
+        String expectedChangePasswordErrorAtTop = "Password change failed. Please correct the errors and try again.";
+
+        LogHelper.info("Get error message next to confirm password field");
+        String actualConfirmPasswordErrorNextTo = changePasswordPage.getConfirmPasswordErrorMessageNextTo();
```
Comment (MinhQuan119) [inline src/test/java/tests/RegisterTests.java]: This variable can be changed to actualConfirmPasswordErrorMsg

Comment (HoanVanHuynh) [reply src/test/java/tests/RegisterTests.java]: Thank you so much

File: src/test/java/tests/RegisterTests.java (hunk 3 above)
Comment (HoanVanHuynh) [reply src/test/java/tests/RegisterTests.java]: Thank you so much

**Review by MinhQuan119 — APPROVED** [review]

**Review by MinhQuan119 — APPROVED** [review]

File: src/test/java/tests/RegisterTests.java [hunk 5]
```diff
@@ -40,4 +44,42 @@ public void tc07_RegisterWithValidInformation() {
         LogHelper.info("Verify that new account is created and success message appears properly");
         Assert.assertEquals(actualResult, expectedResult, "Register success message displays incorrectly");
     }
+
```
Comment (anhlvt) [inline src/test/java/tests/RegisterTests.java:51]: What happen when we run all test class one time?

Comment (HoanVanHuynh) [reply src/test/java/tests/RegisterTests.java:51]: Thank you for the question, in my case it'll run without error

Comment (anhlvt) [reply src/test/java/tests/RegisterTests.java:51]: I don't think so. Please check again with register account

File: src/main/java/page_objects/BasePage.java (hunk 1 above)
Comment (tanmle) [reply src/main/java/page_objects/BasePage.java]: how about tabRegister?

File: src/test/java/tests/RegisterTests.java (hunk 5 above)
Comment (HoanVanHuynh) [reply src/test/java/tests/RegisterTests.java:51]: I have already run two test cases inside my registerTests class at the same without error.
image

File: src/test/java/tests/RegisterTests.java [hunk 6]
```diff
@@ -6,12 +6,16 @@
...
 import page_objects.RegisterPage;
 
 public class RegisterTests extends BaseTest {
 
     private Account account = new Account();
     private RegisterPage registerPage = new RegisterPage();
+    private ChangePasswordPage changePasswordPage = new ChangePasswordPage();
+    private LoginPage loginPage = new LoginPage();
     private String email = DataHelper.getRandomEmail();
     private String password = DataHelper.getRandomPassword(10);
```
Comment (tanmle) [inline src/test/java/tests/RegisterTests.java]: I missed this but we don't need to random password

File: src/test/java/tests/RegisterTests.java (hunk 5 above)
Comment (HoanVanHuynh) [reply src/test/java/tests/RegisterTests.java:51]: So please let me know where I should check to find error, thank you so much

File: src/test/java/tests/RegisterTests.java (hunk 6 above)
Comment (HoanVanHuynh) [reply src/test/java/tests/RegisterTests.java]: thank you so much, I'll change that

File: src/test/java/tests/RegisterTests.java (hunk 5 above)
Comment (anhlvt) [reply src/test/java/tests/RegisterTests.java:51]: Luckily it passed but the TC02 is using data of TC01 to run. How we can create account for TC02 if we use the same email with TC01??

Comment (HoanVanHuynh) [reply src/test/java/tests/RegisterTests.java:51]: thank you so much for your explanation. I think that I'm using precondition method to create data for two tc07 and tc09, so each Tc has own separated data which means tc07 has own account data and tc09 has own account data.

Comment (anhlvt) [reply src/test/java/tests/RegisterTests.java:51]: Did you debug at register account on TC02 and see what happen?

Comment (HoanVanHuynh) [reply src/test/java/tests/RegisterTests.java:51]: I'm so sorry, I had a big mistake, I should put beforemethod instead of beforetest

Comment (anhlvt) [reply src/test/java/tests/RegisterTests.java:51]: I think @BeforeMethod is good but you need to put this `private String email = DataHelper.getRandomEmail();` on pre-condition

File: src/main/java/page_objects/BasePage.java (hunk 1 above)
Comment (HoanVanHuynh) [reply src/main/java/page_objects/BasePage.java]: thank you so much, I'll change tabRegister a little bit.

File: src/test/java/tests/RegisterTests.java (hunk 5 above)
Comment (HoanVanHuynh) [reply src/test/java/tests/RegisterTests.java:51]: thank you so much, but I'm so sorry I do not understand:  If I put private String email = DataHelper.getRandomEmail() into precondition method, I can not reuse email inside other tc

Comment (anhlvt) [reply src/test/java/tests/RegisterTests.java:51]: You can declare a  email var outside and use Random inside pre-condition:
`
private String email;
public void precondition() {
         email = DataHelper.getRandomEmail()
    }`

Comment (HoanVanHuynh) [reply src/test/java/tests/RegisterTests.java:51]: thank you so much, I understood , oh I'm so stupid

**Review by anhlvt — APPROVED** [review]

File: src/test/java/tests/RegisterTests.java (hunk 5 above)
Comment (HoanVanHuynh) [reply src/test/java/tests/RegisterTests.java:51]: Each Tc inside my registertests has own separated data such as email, pid. Thank you so much Mr. Anh

**Pull request merged**